servers/src/main/java/tachyon/worker/block/BlockDataManager.java

```diff
@@ -245,34 +250,21 @@ public void commitBlock(long sessionId, long blockId)
    * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
    * @throws IOException if blocks in eviction plan fail to be moved or deleted
    */
-  public String createBlock(long sessionId, long blockId, int tierAlias, long initialBytes)
-      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException {
-    BlockStoreLocation loc =
-        tierAlias == -1 ? BlockStoreLocation.anyTier() : BlockStoreLocation.anyDirInTier(tierAlias);
-    TempBlockMeta createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
-    return createdBlock.getPath();
-  }
-
-  /**
-   * Creates a block. This method is only called from a data server.
-   *
-   * Call {@link #getTempBlockWriterRemote(long, long)} to get a writer for writing to the block.
-   *
-   * @param sessionId The id of the client
-   * @param blockId The id of the block to be created
-   * @param tierAlias The alias of the tier to place the new block in, -1 for any tier
-   * @param initialBytes The initial amount of bytes to be allocated
-   * @throws IllegalArgumentException if location does not belong to tiered storage
-   * @throws BlockAlreadyExistsException if blockId already exists, either temporary or committed,
-   *         or block in eviction plan already exists
-   * @throws WorkerOutOfSpaceException if this Store has no more space than the initialBlockSize
-   * @throws IOException if blocks in eviction plan fail to be moved or deleted
-   */
-  public void createBlockRemote(long sessionId, long blockId, int tierAlias, long initialBytes)
-      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, IOException {
-    BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(tierAlias);
-    TempBlockMeta createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
-    FileUtils.createBlockPath(createdBlock.getPath());
+  public String createBlock(long sessionId, long blockId, long initialBytes)
+      throws BlockAlreadyExistsException, WorkerOutOfSpaceException, BlockDoesNotExistException,
+      IOException, InvalidWorkerStateException {
+    // add flag check here?
+    BlockStoreLocation loc = BlockStoreLocation.anyDirInTier(StorageLevelAlias.MEM.getValue());
+    TempBlockMeta createdBlock;
+    try {
+      createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, loc, initialBytes);
+      return createdBlock.getPath();
+    } catch (WorkerOutOfSpaceException e) {
+      LOG.info("MEM tier is full, try to create block in any tier");
+      createdBlock = mBlockStore.createBlockMeta(sessionId, blockId, BlockStoreLocation.anyTier(),
```

Comment:
Do we need to put this in catch?Under `anyTier`, wouldn't createBlockMeta perform a best-effort try from the top-most tier?